hoodie-bench/src/main/java/com/uber/hoodie/integrationsuite/converter/InsertToUpdateConverter.java

```diff
@@ -0,0 +1,54 @@
+/*
+ *  Copyright (c) 2019 Uber Technologies, Inc. (hoodie-dev-group@uber.com)
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *           http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ */
+
+package com.uber.hoodie.integrationsuite.converter;
+
+import com.uber.hoodie.integrationsuite.generator.LazyRecordGeneratorIterator;
+import com.uber.hoodie.integrationsuite.generator.UpdateGeneratorIterator;
+import com.uber.hoodie.utilities.sources.converter.Converter;
+import java.util.List;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.spark.api.java.JavaRDD;
+
+/**
+ * This converter creates an update {@link GenericRecord} from an insert/new {@link GenericRecord}
+ */
+public class InsertToUpdateConverter implements Converter<GenericRecord, GenericRecord> {
+
+  private final String schemaStr;
+  // The fields that should not be mutated when converting the insert record to an update record, generally the
+  // record_key
+  private final List<String> partitionPathFields;
+  private final List<String> recordKeyFields;
+  private final int minPayloadSize;
+
+  public InsertToUpdateConverter(String schemaStr, int minPayloadSize, List<String> partitionPathFields,
```

Comment:
I don't quite follow how we decide this key has been already inserted in a previous batch? do we track all keys inside the dataset somewhere? what am I missing?